ionoscloud/resource_autoscaling_group.go

```diff
@@ -0,0 +1,484 @@
+									},
+									"user_data": {
+										Optional:    true,
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: "User-data (Cloud Init) for this replica volume.",
+									},
+									"image_password": {
+										Optional:    true,
+										Type:        schema.TypeString,
+										Description: "Image password for this replica volume.",
+									},
+									"boot_order": {
+										Optional: true,
+										Type:     schema.TypeString,
+										Description: `Determines whether the volume will be used as a boot volume. Set to NONE, the volume will not be used as boot volume. 
+Set to PRIMARY, the volume will be used as boot volume and set to AUTO will delegate the decision to the provisioning engine to decide whether to use the volume as boot volume.
+Notice that exactly one volume can be set to PRIMARY or all of them set to AUTO.`,
+									},
+									"bus": {
+										Optional:    true,
+										Type:        schema.TypeString,
+										Default:     autoscaling.BUSTYPE_VIRTIO,
+										Description: `The bus type of the volume. Default setting is 'VIRTIO'. The bus type 'IDE' is also supported.`,
+									},
+									"backup_unit_id": {
+										Type:        schema.TypeString,
+										Description: "The uuid of the Backup Unit that user has access to. The property is immutable and is only allowed to be set on a new volume creation. It is mandatory to provide either 'public image' or 'imageAlias' in conjunction with this property.",
+										Optional:    true,
+										Computed:    true,
+									},
+								}},
+						},
+					},
+				},
+			},
+			"datacenter_id": {
+				Type:             schema.TypeString,
+				Description:      "Unique identifier for the resource",
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Description: "Location of the data center.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceAutoscalingGroupCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	group, err := autoscalingService.GetAutoscalingGroupDataCreate(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured getting data from the provided schema: %w", err))
+	}
+	if group.Properties != nil && group.Properties.Name != nil {
+		log.Printf("[DEBUG] autoscaling group data extracted: %+v", *group.Properties.Name)
+	}
+
+	autoscalingGroup, _, err := client.CreateGroup(ctx, *group)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("error creating autoscaling group: %w", err))
+		return diags
+	}
+
+	d.SetId(*autoscalingGroup.Id)
+	log.Printf("[INFO] autoscaling Group created. Id set to %s", *autoscalingGroup.Id)
```

Comment:
@cristiGuranIonos let's keep these log messages consistent by using either capital letters for `group` or not.